### ISSUE TITLE
Reload recommended jobs after candidate likes one of them

### ISSUE THREAD

**Issue by lacogubik** [WorksHub/client#6]
## Problem
Liked jobs are removed from recommendations, so if the user reloads page manually the liked job will be removed. 

## Implementation
This should happen in both the dashboard (`/`) and the recommended jobs page (`/recommended`).

## Acceptance Criteria
When candidate clicks on heart icon on the job in recommended section of the dashboard or in recommended job page, section or page will reload.

## Notes
We already trigger reload when user blacklists (clicks on close icon) a job, so we just need to copy that behaviour.

**Comment by cmnstmntmn** [comment]
i belive a a flash notification would be a nice to have for 
cases where the element is removed from the UI.

i've made a proof of concept [here](https://flems.io/#0=N4IgzgpgNhDGAuEAmIBcIB0ALeBbKIANCAGYCWMYaA2qAHYCGuEamO+RIsA9nYn6wA8AIwCu8eLwAEsKAzBgAvAB0QABwC0AZlVTessrADWKkEm6xRzPhgDmEeAFEY1+ACEAngEkkACgDkJHJgWP4AlBiy8mAAMmRg8BiStrYwAUh04aoAfMp0UlKAfBuAh7tSzmQAHmQATlJIEABu0NxqENV5ggD0YhK8uXQdSGQNUmRIpkHyWLpRChPBWFLCthrc1Qx09lIA7lhkiFKaWnUDIP0FAMJYcEZLDMZSGx7wZMxSggnVvLbZ5VW19SaUBabS6n2+2Skojo9VqJSkUDIRmQUkgCDIvAwHU6Qwa2U4kBg6N4VHQAFZUAAmACMIAAvoR6EwWOgMAArKjEHh8CACdCcHi4NQUNqsYQMYTQemMkCMZisSIKAW8fjwBXLVbrTYQKTAPIFcXGWxfaFIDQ8YHVVBSaq2YS+SlksmEKTUykAFhd1OpAAYwgU8nS8nkMLt9jq9fkZNxLdaw4gANyB4N0DBHXX6w4MJBDTbWjAADmqEFwSboQYGqYyGajQzAajkHmtdF4EDLFZDkxCNYKBTU3DA+wxdGt5AqyDLvaW3F6uFQPsnveqZFsOHni4KG1eDBevGtvMQ1Q0JC+uA0whnklwUgwWjAUgg8ggLogVXgGnM23yGEp98fkCkd0MDJe9qSkFtqlwBgoCkEg1m2BhqiQe9qmhOgyE2DcdgoKBzSwDZ7GteAtTAODIPbFMAAFkQ8E9mT-HkjxPbgzwvWce17Zjr0jKcCmIjZSLWOd+LoMA5EQABNXxfR9ABSMIsIrKdJA45SSLI4SSPEiApL9RTMyUqQ8moiBaPWZg-zfD9uC-VTYNPOy+PUoTUBEsSdx03w9MzApDL47hHKkNyNNcrSPKk6kAHY5IUnypArCsCWgOBd1E1hqS0VBvXpABdOkgA)

**Comment by acron0** [comment]
> i belive a a flash notification would be a nice to have for
> cases where the element is removed from the UI.
> 
> i've made a proof of concept [here](https://flems.io/#0=N4IgzgpgNhDGAuEAmIBcIB0ALeBbKIANCAGYCWMYaA2qAHYCGuEamO+RIsA9nYn6wA8AIwCu8eLwAEsKAzBgAvAB0QABwC0AZlVTessrADWKkEm6xRzPhgDmEeAFEY1+ACEAngEkkACgDkJHJgWP4AlBiy8mAAMmRg8BiStrYwAUh04aoAfMp0UlKAfBuAh7tSzmQAHmQATlJIEABu0NxqENV5ggD0YhK8uXQdSGQNUmRIpkHyWLpRChPBWFLCthrc1Qx09lIA7lhkiFKaWnUDIP0FAMJYcEZLDMZSGx7wZMxSggnVvLbZ5VW19SaUBabS6n2+2Skojo9VqJSkUDIRmQUkgCDIvAwHU6Qwa2U4kBg6N4VHQAFZUAAmACMIAAvoR6EwWOgMAArKjEHh8CACdCcHi4NQUNqsYQMYTQemMkCMZisSIKAW8fjwBXLVbrTYQKTAPIFcXGWxfaFIDQ8YHVVBSaq2YS+SlksmEKTUykAFhd1OpAAYwgU8nS8nkMLt9jq9fkZNxLdaw4gANyB4N0DBHXX6w4MJBDTbWjAADmqEFwSboQYGqYyGajQzAajkHmtdF4EDLFZDkxCNYKBTU3DA+wxdGt5AqyDLvaW3F6uFQPsnveqZFsOHni4KG1eDBevGtvMQ1Q0JC+uA0whnklwUgwWjAUgg8ggLogVXgGnM23yGEp98fkCkd0MDJe9qSkFtqlwBgoCkEg1m2BhqiQe9qmhOgyE2DcdgoKBzSwDZ7GteAtTAODIPbFMAAFkQ8E9mT-HkjxPbgzwvWce17Zjr0jKcCmIjZSLWOd+LoMA5EQABNXxfR9ABSMIsIrKdJA45SSLI4SSPEiApL9RTMyUqQ8moiBaPWZg-zfD9uC-VTYNPOy+PUoTUBEsSdx03w9MzApDL47hHKkNyNNcrSPKk6kAHY5IUnypArCsCWgOBd1E1hqS0VBvXpABdOkgA)

Flash notifications are a good idea, however in this case it's more about a functionality adjustment rather than a visual notification.

**Comment by eckardjf** [comment]
I think I have this completed, but I'm having some trouble with testing. There are no recommended jobs in the seed data and I'm having trouble creating new roles (as company and admin).

**Comment by Alex-Bakic** [comment]
# Mini walkthrough of the reloading issue... 

#### question at the bottom...

For the moment , with the like button on job cards, when they are clicked they do not trigger reloads of the recommended or dashboard pages, contrary to the blacklist button. What blacklist does is take the job out of the sub-db path `[:wh.logged-in.personalised-jobs.db/sub-db ▶{...,  :wh.logged-in.personalised-jobs.db/jobs [{} , ... ,  {}]}]` and puts it in the blacklisted `▶{:wh.user.db/sub-db {:wh.user.db/blacklist #{"remote-junior-maker-of-everything-dd48b"}}}` section. `::toggle-job-like` does the same, but the difference being blacklist will recall either the dashboard or the recommended page to show that it has been removed:

    (reg-event-fx
      ::blacklist-job
      db/default-interceptors
      (fn [{db :db} [{:keys [id] :as job} action]]
          {:graphql         {:query      blacklist-job-mutation
                             :variables  {:id id}
                             ;; in the job.cljc file, on-close is passed into blacklist, allowing it to close the job
                             ;; on the relevant page.
                             :on-success (if (= action :reload-recommended)
                                            [:personalised-jobs/fetch-jobs-by-type :recommended 1]
                                            ;; need to refer by full name since it's in another module
                                            [:wh.logged-in.dashboard.events/fetch-recommended-jobs])}
           :analytics/track ["Job Blacklisted" job]}))

So in the case that graphql manages to mutate the persistent storage, which will be reflected in re-frame, we'll recall the approriate page. So by adding the `on-close` from `job.cljc` we should be able to add this functionality into the like event:

    ;; event which moves the job into the liked section, to be shown in the liked page...
    (reg-event-db
      ::toggle-job-like-success
      db/default-interceptors
      (fn [db [id]]
          (update-in db [:wh.user.db/sub-db :wh.user.db/liked-jobs] util/toggle id)))

    (reg-event-fx
      ::toggle-job-like
      db/default-interceptors
      (fn [{db :db} [{:keys [id] :as job} action]]
         (let [liked (contains? (get-in db [:wh.user.db/sub-db :wh.user.db/liked-jobs]) id)
              event-name (str "Job " (if liked "Removed Like" "Liked"))]
           {:graphql         {:query      like-job-mutation
                              :variables  {:id id, :add (not liked)}
                              :on-success [::toggle-job-like-success id]}
            :analytics/track [event-name job]

            :dispatch (if (= :reload-recommended action)
                        [:personalised-jobs/fetch-jobs-by-type :recommended 1]
                        [:wh.logged-in.dashboard.events/fetch-recommended-jobs])})))

We use dispatch instead of plugging it into `on-success` as we need to be able to toggle the job in and out of the liked jobs page, which needs to be handled by a separate event. This is also important as we want to make sure that the job is pulled out before we start re-rendering pages. 

## Issue

Now when we go to the liked page, and click to "unlike" it will re-render and you will no longer see the job, which is ideal; however, for the other way round , clicking "like" in the dashboard doesn't work as `if` will evaluate for the case of `:reload-recommended`. Flipping doesn't solve the problem, as we get the inverse of the issue, but also another facet of the issue is that job cards will be presented in many different parts of the app, under the blog posts, under company profiles etc. But say we don't want jobs to re-render for these cases, we should use `case` to allow for liking and unliking to work appropriately.

The only issue is I only know one of the keywords, seen above, but what are the keywords that correspond to each action, for dashboard and for the personalised section?
